### DISCUSSION TITLE
TypeError: Object of type File is not JSON serializable

### DISCUSSION THREAD

**Discussion by UnsuccessfullHTTP** [Rapptz/discord.py#5872] [Q&A] [answered]
Im making a bot that detects when you add a reaction to a message and edits it with an embed that has an image inside.
When it tries to edit the message it returns `TypeError: Object of type File is not JSON serializable`
How can I fix it?
This is the part of the code that I used to edit the message :
```
embed.set_image(url="attachment://image.png")
file = discord.File("imgrender/render.png", filename="image.png")
await reaction.message.edit(file=file, embed=embed)
```

**Comment by AbstractUmbra** [comment, marked as answer]
You cannot edit the attachments of a Message.
